src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,742 @@
+                      [0, drawerWidth, drawerWidth + 1],
+                      [0, drawerWidth, drawerWidth]
+                    )
+                  : interpolate(
+                      event.x - containerWidth,
+                      [-drawerWidth - 1, -drawerWidth, 0],
+                      [drawerWidth, drawerWidth, 0]
+                    )
+                : 0,
+              sideCorrection *
+                (event.translationX +
+                  (drawerOpened ? drawerWidth * -gestureOrientation : 0))
+            );
+
+          drawerTranslation.value =
+            drawerType === DrawerType.FRONT
+              ? sideCorrectedDragX.value +
+                interpolate(
+                  -1 * sideCorrectedDragX.value,
+                  [drawerWidth - 1, drawerWidth, drawerWidth + 1],
+                  [0, 0, 1]
+                )
+              : sideCorrectedDragX.value;
+        })
+        .onEnd(handleRelease);
+    }, [
+      dragX,
+      drawerLockMode,
+      drawerTranslation,
+      drawerType,
+      drawerWidth,
+      emitStateChanged,
+      gestureOrientation,
+      handleRelease,
+      edgeHitSlop,
+      fillHitSlop,
+      minSwipeDistance,
+      props.hideStatusBar,
+      props.keyboardDismissMode,
+      sideCorrectedDragX.value,
+      overlayDismissGesture,
+      drawerOpened,
+      isFromLeft,
+      containerWidth,
+      sideCorrection,
+      drawerState,
+      activeCursor,
+      enableTrackpadTwoFingerGesture,
+      mouseButton,
+      statusBarAnimation,
+    ]);
+
+    // When using RTL, row and row-reverse flex directions are flipped.
+    const reverseContentDirection = I18nManager.isRTL
+      ? isFromLeft
+      : !isFromLeft;
+
+    const dynamicDrawerStyles = {
+      backgroundColor: drawerBackgroundColor,
+      width: drawerWidth,
+    };
+
+    const containerStyles = useAnimatedStyle(() => {
+      if (drawerType === DrawerType.FRONT) {
+        return {};
+      }
+
+      return {
+        transform: [
+          {
+            translateX: interpolate(
+              openValue.value,
+              [0, 1],
+              [0, drawerWidth * sideCorrection],
+              Extrapolation.CLAMP
+            ),
+          },
+        ],
+      };
+    });
+
+    const drawerAnimatedStyle = useAnimatedStyle(() => {
+      const closedDrawerOffset = drawerWidth * -sideCorrection;
+
+      return {
+        transform: [
+          {
+            translateX:
+              drawerType === DrawerType.BACK
+                ? 0
+                : drawerState === DrawerState.IDLE
+                ? drawerOpened
+                  ? 0
+                  : closedDrawerOffset
+                : interpolate(
+                    openValue.value,
+                    [0, 1],
+                    [closedDrawerOffset, 0],
+                    Extrapolation.CLAMP
+                  ),
+          },
+        ],
+        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+      };
```

Comment:
Same here, is there a way to rewrite it? For example:


```suggestion
      let translation;

      if (drawerType == DrawerType.BACK) {
        translation = 0;
      } else if (drawerState === DrawerState.IDLE) {
        translation = drawerOpened ? 0 : closedDrawerOffset;
      } else {
        translation = interpolate(
          openValue.value,
          [0, 1],
          [closedDrawerOffset, 0],
          Extrapolation.CLAMP
        );
      }

      return {
        transform: [{ translateX: translation }],
        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
      };
```

Reply:
Rewritten in [be4979c](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/be4979cf2b01874ddd925e2c47f812ed4482be29)